relay-dynamic-config/src/project.rs

```diff
@@ -47,6 +57,10 @@ pub struct ProjectConfig {
     /// Maximum sampled event retention for the organization.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub downsampled_event_retention: Option<u16>,
+    /// Standard and down sampled event retentions per DataCategory.
+    /// The key is the string DataCategory.name
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub retentions: Option<HashMap<String, RetentionSettings>>,
```

Comment:
For consistency with other maps in our config and https://develop.sentry.dev/ingestion/relay/relay-best-practices/#data-structures:

```suggestion
    pub retentions: Option<BTreeMap<String, RetentionSettings>>,
```